src/sentry/migrations/0645_backfill_add_uuid_to_all_rule_actions.py

```diff
@@ -0,0 +1,55 @@
+# Generated by Django 5.0.2 on 2024-02-13 20:45
+from django.db import migrations
+
+from sentry.api.serializers.rest_framework import ensure_action_uuid
+from sentry.new_migrations.migrations import CheckedMigration
+
+
+def add_uuid_to_all_rule_actions(apps, schema_editor) -> None:
+    """
+    Backfill all rule actions with an uuid.
+    If the rule's data does not have any actions, simply move on and do not update.
+    """
+    Rule = apps.get_model("sentry", "Rule")
+
+    all_rules = Rule.objects.all()
+    for rule in all_rules:
```

Comment:
It's a good idea to use `RangeQuerySetWrapperWithProgressBarApprox` here. It makes sure we don't attempt to pull every row into memory at once.

Reply:
Ohh, this is good to know, thank you!